### PR TITLE
Add controllers and routes

### PR DESCRIPTION
**Pull request by Surafels**
Hello 👋

In this branch, the following are implemented:
For each URL the following are created:

- A route.
- An action in the correct controller.
- A view file.

View files  include only placeholders written in plain HTML

**rails routes output 👇**
image

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+get 'users', to: 'users#index', as: 'users'
+get 'users/:id', to: 'users#show', as: 'user'
   # Reveal health status on /up that returns 200 if the app boots with no exceptions, otherwise 500.
   # Can be used by load balancers and uptime monitors to verify that the app is live.
   get "up" => "rails/health#show", as: :rails_health_check
```
Comment (Ghiftee) [inline config/routes.rb]: - [ ] You have done a great job defining these routes as stated in the project requirement. There is just one little thing left on the way to having an approved project 🎉 Kindly ensure that your route definition has a RESTful URL structure. Following RESTful principles makes your API or application more predictable and consistent. It is also considered best practice. Using RESTful URL structure would mean that your routes would be defined like this:

```
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```
  
- The resources :users block inside resources :posts signifies a nested resource relationship. This adheres to RESTful conventions where posts are nested under users, indicating that posts are related to and dependent on users.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Surafels 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ Project was created without `rails generate scaffold` command 
✔️ `Index` and `Show` actions are present for users and posts
✔️ View files and routes are present

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review 👇🏽 _

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by danmainah — APPROVED** [review]
Hi @Surafels , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the changes requested to ensure the routes are  RESTful.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.htm.erb [hunk 2]
```diff
@@ -0,0 +1 @@
+<h1>Here is the list of posts for the user:</h1>
```
Comment (danmainah) [inline app/views/posts/show.htm.erb]: - [ ] [optional] Kindly consider fixing the error of your show page from ` show.htm.erb` to ` show.html.erb`

**Pull request merged**